dialog/fileicon.go

```diff
@@ -97,9 +97,8 @@ func (s fileIconRenderer) Refresh() {
 	canvas.Refresh(s.item)
 }
 
-func mimeTypeGet(path string) (ext, mimeType, mimeSubType string) {
-	uri := storage.NewURI(path)
-	ext = uri.Extension()
+func mimeTypeGet(uri fyne.URI) (mimeType, mimeSubType string) {
```

Comment:
It is odd having 'Get' as a suffix. Maybe "uriToMime" or "mimeTypeFromURI"?

Reply:
the mimeTypeFromURI is uri.MimeType() ... It already exists as discussed on Tilo's comment earlier. uri.MimeType() returns a full mimetype: "image/jpeg" for instance... For the purposes of filtering and building out the file icon however we need it split so that we have "image" and "jpeg" ... This isn't a public method or anything - just a utility to avoid duplication by splitting the mimetype returned from URIs.  I'm open for suggestions here.

Reply:
Perhaps mimeTypeSplit(uri fyne.URI) (mimetype, mimeSubType  string) would make sense

Reply:
Again the verb suffix is odd, "splitMime"?

Reply:
Hangover from C and EFL days I guess - they always put verbs on the end. I would agree that splitMime seems clean.

Reply:
I “unresolved” this as I think Stuart’s point was not responded to. Not sure if GitHub automatically was dismissing or not.

Reply:
Oh I was thinking of split as a noun and not a verb. As in the return is a mime type split. I could see that being confusing and will change it to splitMimeType